### PR TITLE
Added types.

### PR DESCRIPTION
**Pull request by mzhang28**

### CODE REVIEW

File: src/parser/commands/if.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// tslint:disable:max-classes-per-file
+
+import { Expression } from "../expression";
+import { Node } from "../node";
+import { ElifNode } from "./elif";
+import { ElseNode } from "./else";
+import { Command, process } from "./index";
+import { CommandNode } from "./node";
+
+export class IfCommand extends Command {
```
Comment (ScottSWu) [inline src/parser/commands/if.ts:10]: I'll take care of this in `src/commands`

File: src/parser/index.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { IllegalArgumentException, IllegalStateException } from "./errors";
+import { Node } from "./node";
+import { Resource, Resources } from "./resources";
+import { Scene } from "./scene";
+import { Section } from "./section";
+
+export class AbstractSyntaxTree extends Node {
```
Comment (ScottSWu) [inline src/parser/index.ts:7]: This should probably be in its own file.
Typically `index.ts` is for indexing the folder,
```
export { Node } from "./node";
export { Scene } from "./scene";
...
```

File: src/parser/commands/if.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public command(): IfCommand {
+        let falseNode: Node = new CommandNode();
+        if (this.falseNode !== undefined) {
+            falseNode = this.falseNode;
+        }
+        const command: IfCommand = new IfCommand(this.condition, this, falseNode);
+        return command;
+    }
+    public consumeLine(line: string): void {
+        const command = process(line, this);
```
Comment (ScottSWu) [inline src/parser/commands/if.ts:48]: Should this be `this.process`?

File: src/parser/index.ts (hunk 2 above)
Comment (mzhang28) [reply src/parser/index.ts:7]: ok i'll change that

File: src/parser/commands/if.ts (hunk 3 above)
Comment (mzhang28) [reply src/parser/commands/if.ts:48]: no? `process` isn't a property of IfNode

Comment (ScottSWu) [reply src/parser/commands/if.ts:48]: Ooo I see the import at the top now. Maybe move process as a static function within CommandNode, or Node.

File: src/parser/commands/if.ts (hunk 1 above)
Comment (ScottSWu) [reply src/parser/commands/if.ts:10]: Jk ignore this.

File: src/parser/commands/if.ts (hunk 3 above)
Comment (mzhang28) [reply src/parser/commands/if.ts:48]: i tried doing process as a static function of CommandNode, but that raised circular import problems

**Pull request closed without merging**